eva-usage-stats/ftp_usage.py

```diff
@@ -0,0 +1,149 @@
+        b['ip2location']['city'],
+        b['ip2location']['domain'],
+        b['ip2location']['isp'],
+        b['ip2location']['usage_type'],
+    ) for b in batch]
+    with get_metadata_connection_handle('development', private_config_xml_file) as metadata_connection_handle:
+        with metadata_connection_handle.cursor() as cursor:
+            query_insert = (
+                'INSERT INTO eva_web_srvc_stats.ftp_traffic '
+                'VALUES (%s, cast(%s as timestamp with time zone), %s, %s, %s, %s, '
+                'cast(%s as timestamp without time zone), %s, %s, %s, %s, %s, %s, '
+                '%s, %s, %s, %s, %s, %s, %s)'
+            )
+            psycopg2.extras.execute_batch(cursor, query_insert, rows)
+
+
+def get_most_recent_timestamp(private_config_xml_file):
+    with get_metadata_connection_handle('development', private_config_xml_file) as metadata_connection_handle:
+        results = get_all_results_for_query(
+            metadata_connection_handle,
+            "select max(event_ts) as recent_ts from eva_web_srvc_stats.ftp_traffic;"
+        )
+        if results and results[0][0]:
+            return results[0][0].timestamp()
+    return None
+
+
+@retry(tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+def query(kibana_host, basic_auth, private_config_xml_file, batch_size):
+    first_query_url = os.path.join(kibana_host, 'ftplogs*/_search?scroll=24h')
+    query_conditions = [{'query_string': {'query': 'file_name:("/pub/databases/eva/")'}}]
+    most_recent_timestamp = get_most_recent_timestamp(private_config_xml_file)
+    if most_recent_timestamp:
+        query_conditions.append({'range': {'@timestamp': {'gt': most_recent_timestamp}}})
+    post_query = {
+        'size': str(batch_size),
+        'query': {'bool': {'must': query_conditions}}
+    }
+
+    response = requests.post(first_query_url, auth=basic_auth, json=post_query)
+    response.raise_for_status()
+    data = response.json()
+    total = data['hits']['total']
+    if total == 0:
+        logger.info('No results found')
+        return None, None, None
+
+    scroll_id = data['_scroll_id']
+    return scroll_id, total, data['hits']['hits']
+
+
+@retry(tries=4, delay=2, backoff=1.2, jitter=(1, 3))
+def scroll(kibana_host, basic_auth, scroll_id):
+    query_url = os.path.join(kibana_host, '_search/scroll')
+    response = requests.post(query_url, auth=basic_auth, json={'scroll': '24h', 'scroll_id': scroll_id})
+    response.raise_for_status()
+    data = response.json()
+    return data['hits']['hits']
+
+
+def main():
+    parser = ArgumentParser(description='Retrieves data from Kibana and dumps into a local postgres instance')
+    parser.add_argument('--kibana-host', help='Kibana host to query, e.g. http://example.ebi.ac.uk:9200', required=True)
+    parser.add_argument('--kibana-user', help='Kibana API username', required=True)
+    parser.add_argument('--kibana-pass', help='Kibana API password', required=True)
```

Comment:
Is this a common password or user-specific? If it is the latter, I am not sure if we should expose it on the command line...

Reply:
It is the same (weak) password for everyone

Reply:
:+1: